packages-next/keystone/src/lib/createKeystone.ts

```diff
@@ -70,8 +70,16 @@ export function createKeystone(config: KeystoneConfig): Keystone {
       itemQueryName: listConfig.graphql?.itemQueryName,
       hooks: listConfig.hooks,
     } as any) as any;
+    // Default the labelField to `name`, `label`, or `title` if they exist; otherwise fall back to `id`
     const labelField =
-      (listConfig.ui?.labelField as string | undefined) ?? (listConfig.fields.name ? 'name' : 'id');
+      (listConfig.ui?.labelField as string | undefined) ??
+      (listConfig.fields.name
+        ? 'name'
+        : listConfig.fields.label
+        ? 'label'
+        : listConfig.fields.title
+        ? 'title'
+        : 'id');
```

Comment:
Can we hoist this logic up into core Keystone instead of having it here?

Reply:
Nope, because core Keystone doesn't have a concept of a `labelField` option. It has a label resolver, which we could update to match this logic as well, but that's a breaking change for the current system (and potentially a big one, to change the default behaviour of the `_label_` resolver) so imo that's out of scope for this PR.